webpack.dev.ts

```diff
@@ -1,21 +1,20 @@
-export {}
-const path = require('path')
-const merge = require('webpack-merge')
-const common = require('./webpack.common.ts')
+const pathDev = require('path')
+const mergeDev = require('webpack-merge')
+const commonDev = require('./webpack.common.ts')
 const PORT = parseInt(process.env.PORT, 10) || 8080
 const PUBLIC = process.env.PUBLIC || undefined
-const {BASE_PATH} = require('./src/utils/env')
-const BundleAnalyzerPlugin = require('webpack-bundle-analyzer')
-  .BundleAnalyzerPlugin
+const BASE_PATH_DEV = require('./src/utils/env').BASE_PATH
```

Comment:
😳

Reply:
are the `pathDev`, `mergeDev`, and `commonDev` name changes related to naming collisions also?

What's the issue / error message being thrown here? If there are naming collisions, do the config files downstream of `webpack.common.ts` even need to do these imports, since it seems to imply these files colliding are already in scope? Can we just use the references already imported by `webpack.common.ts`?

Reply:
>How do you propose we do that?

@asalem1 If they're already in scope, then you should be able to just refer to them without importing them. But I don't think that will work here, since these files are used independently of each other. It seems pretty clear that TypeScript is using them in a way that we don't normally use them.

It looks like the `export {}` was meant as a [fix for this issue.](https://stackoverflow.com/a/50913569/92446) Does that no longer work?

All of this makes me wonder what value we get type checking our webpack files. My feeling is: who cares if they're not valid TypeScript, they're not being run like production code, and type errors in build config files will either be fixed if they cause a problem, or they don't matter. (The same for e2e test files actually, we're not even writing TS when using Cypress' DSL, so who cares if Cypress test files are valid TS?)

I'm also very confused what the past intent towards type checking webpack config files was. [We ignore `webpack.*.ts`](https://github.com/influxdata/ui/blob/master/tsconfig.json#L39) in `tsconfig.json`, but then we also have a [`webpack.tsconfig.json`](https://github.com/influxdata/ui/blob/master/webpack.tsconfig.json) file.

@rockstar what are your thoughts on type checking webpack files?

Reply:
> My feeling is: who cares if they're not valid TypeScript, they're not being run like production code, and type errors in build config files will either be fixed if they cause a problem, or they don't matter.

In a project that had well-defined types, I might have a different opinion. I know there were conversations a few months ago where we talked about making stronger types, and I think we should. When it comes to the top 5 biggest priorities in our application though, I don't think that's the best use of our time. I think "upgrade TypeScript to a version that isn't 3 years old" is a higher priority than strictly typed build-related files.